test/utils/fizzy_c_engine.cpp

```diff
@@ -0,0 +1,125 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <fizzy/fizzy.h>
+
+#include <test/utils/adler32.hpp>
+#include <test/utils/wasm_engine.hpp>
+#include <cassert>
+#include <cstring>
+
+namespace fizzy::test
+{
+class FizzyCEngine final : public WasmEngine
+{
+    std::unique_ptr<FizzyInstance, void (*)(FizzyInstance*)> m_instance = {
+        nullptr, fizzy_free_instance};
+
+public:
+    bool parse(bytes_view input) const final;
+    std::optional<FuncRef> find_function(
+        std::string_view name, std::string_view signature) const final;
+    bool instantiate(bytes_view wasm_binary) final;
+    bool init_memory(bytes_view memory) final;
+    bytes_view get_memory() const final;
+    Result execute(FuncRef func_ref, const std::vector<uint64_t>& args) final;
+};
+
+namespace
+{
+FizzyExecutionResult env_adler32(
+    void*, FizzyInstance* instance, const FizzyValue* args, size_t, int)
+{
+    auto* memory = fizzy_get_instance_memory_data(instance);
+    assert(memory != nullptr);
+    const auto size = fizzy_get_instance_memory_size(instance);
+    const auto ret = fizzy::test::adler32(
+        bytes_view(memory, size)
+            .substr(static_cast<uint32_t>(args[0].i64), static_cast<uint32_t>(args[1].i64)));
```

Comment:
Why not `.as<uint32_t>()` like `fizzy_engine.cpp` does it?

Reply:
Because these are `FizzyValue`s, they don't have `as()`

Reply:
Ah okay! :)